transform/app/tasks/transform/dump/validate.py

```diff
@@ -16,12 +24,31 @@ def validate_dump(prev_task_status: Optional[CeleryTaskStatus], s3_url: str) ->
     """Task to validate the dump.
     Validation scope:
     - Access,
-    - Dump structure,
-    - Data types inside the dump,
+    - Dump folder structure,
+    - Data types inside the dump, TODO
     - Extension of files"""
     logger.info("Validating dump...")
+
+    s3_client = connect_to_s3(
+        settings.S3_ACCESS_KEY, settings.S3_SECRET_KEY, settings.S3_ENDPOINT
+    )
+
     try:
-        return CeleryTaskStatus(status="toBeImplemented").dict()
+        # 1. Validate access to S3
+        validate_s3_bucket_access(s3_client, s3_url)
+
+        # 2. Validate directory
+        validate_s3_directory_access(s3_client, s3_url)
+
+        # 3. Validate folder structure in the dump
+        validate_s3_directory_structure(s3_client, s3_url)
+
+        # 4. Validate file extensions
+        validate_files_extension(s3_client, s3_url)
+
+        logger.info("Dump validated successfully.")
+
+        return CeleryTaskStatus(status="validated").dict()
```

Comment:
Lets set the status as `success` or `failure` - please set `failure`  when it failed. Each task should assign the same statuses.
Lets make some python file called celery_tasks_statuses and there assign:
SUCCESS =  `success` ...
Later on import those and set inside celery tasks. If you can please refactor other celery tasks to use the same statuses

Reply:
or you can name it successful and unsuccessful - decide on your own :D

Reply:
Added new `task_statuses.py` in `app.services.celery`, with 3 states, which are used when `CeleryTaskStatus` is returned